nomad/job_endpoint.go

```diff
@@ -1412,7 +1412,7 @@ func (j *Job) List(args *structs.JobListRequest, reply *structs.JobListResponse)
 				paginator, err := paginator.NewPaginator(iter, tokenizer, filters, args.QueryOptions,
 					func(raw interface{}) error {
 						job := raw.(*structs.Job)
-						summary, err := state.JobSummaryByID(ws, namespace, job.ID)
+						summary, err := state.JobSummaryByID(ws, job.Namespace, job.ID)
 						if err != nil {
```

Comment:
```suggestion
						if err != nil || summary == nil {
```

Reply:
Ops sorry...it actually seems like it returns `nil, nil` on purpose, so maybe it's possible:
https://github.com/hashicorp/nomad/blob/v1.2.6/nomad%2Fstate%2Fstate_store.go#L2070

Reply:
Right, but we never seem to actually hit this in practice, otherwise we'd be throwing panics all over the place because all the downstream code assumes it's non-nil. If we assert this here, we might find a case where we get spurious errors but we'll prevent those cases from panicking.